WordPress/Classes/Services/NotificationSyncMediator.swift

```diff
@@ -373,8 +432,26 @@ private extension NotificationSyncMediator {
     ///     - noteObjectID: CoreData ObjectID
     ///
     func updateReadStatus(_ status: Bool, forNoteWithObjectID noteObjectID: NSManagedObjectID) {
-        let note = mainContext.loadObject(ofType: Notification.self, with: noteObjectID)
-        note?.read = status
+        updateReadStatus(status, forNotesWithObjectIDs: [noteObjectID])
+    }
+
+    /// Updates the Read status, of an array of Notifications, as specified.
+    ///
+    /// Note: This method uses *saveContextAndWait* in order to prevent animation glitches when pushing
+    /// Notification Details.
+    ///
+    /// - Parameters:
+    ///     - status: New *read* value
+    ///     - notesObjectIDs: CoreData ObjectIDs
+    ///
+    func updateReadStatus(_ status: Bool, forNotesWithObjectIDs notesObjectIDs: [NSManagedObjectID]) {
+        let predicate = NSPredicate(format: "SELF IN %@", notesObjectIDs)
+
+        let notes = mainContext.allObjects(ofType: Notification.self, matching: predicate)
+
+        for note in notes {
+            note.read = status
+        }
```

Comment:
I think using `.forEach` will be a bit more efficient.
`notes.forEach { $0.read = status }`

Reply:
Sure I can do it. As far as I know though, `forEach` runs a `for-in` loop inside so their efficiency should be the same and due to the nature of closure (return vs break) for-ins have been my go-to choice.

Reply:
Ah, ok. I wasn't sure. Either way is fine then.